file-avail-s3.md

```diff
@@ -0,0 +1,235 @@
+written to the job working directory then compressed in to a single
+file before being transferred back to an S3 bucket at the end of the job.
+
+To prepare job-specific data that is large enough
+and exists as multiple files or directories (or a directory of multiple
+files), first create a compressed tar package before placing the file in
+an S3 bucket (either before submitting jobs, or within jobs before
+transferring output to). For example:
+
+``` {: .term}
+$ tar -czvf job_package.tar.gz file_or_dir
+```
+
+## D. Use the Transfer Server
+
+Movement of large data into/out of S3 buckets before and after jobs
+should be performed via CHTC's transfer server, as below, and
+**not via a CHTC submit server.** After obtaining an account on the
+transfer server and creating an S3 bucket, copy relevant files directly into your
+home directory from your own computer:
+
+- Example `scp` command on your own Linux or Mac computer:
+``` {.term}
+$ scp large-input.file username@transfer.chtc.wisc.edu:/home/username/
+```
+- If using a Windows computer:
+	- Using a file transfer application, like WinSCP, directly drag the large
+file from its location on your computer to a location within
+`/home/username/` on transfer.chtc.wisc.edu.
+
+Then in an SSH session on the transfer server, copy files in to your
+S3 bucket:
+
+``` {.term}
+$ mc cp large-input.file chtc/my-bucket
+```
+
+## E. Remove Files After Jobs Complete
+
+As with all CHTC file spaces, data should be removed from S3 buckets AS
+SOON AS IT IS NO LONGER NEEDED FOR ACTIVELY-RUNNING JOBS. Even if it
+will be used again in the future, it should be deleted from and copied
+back at a later date. Files can be taken out of S3 buckets using similar
+mechanisms as uploaded files. In an SSH session on the transfer
+server, copy files from your bucket to your home directory:
+
+``` {.term}
+$ mc cp chtc/my-bucket/large-output.file .
+```
+
+Then copy files from the transfer server to your own computer:
+
+- Example `scp` command on your own Linux or Mac computer:
+``` {.term}
+$ scp username@transfer.chtc.wisc.edu:/home/username/large-output.file .
+```
+- If using a Windows computer:
+	- Using a file transfer application, like WinSCP, directly drag the large
+file from its location within `/home/username/` on
+transfer.chtc.wisc.edu to your computer.
+
+To remove a file inside your S3 bucket, in an SSH session on the
+transfer server:
+
+``` {.term}
+$ mc rm chtc/my-bucket/large-input.file
+$ mc rm chtc/my-bucket/large-output.file
+```
+
+To remove an entire bucket (**only do this if you are certain the
+bucket is no longer needed**):
+
+``` {.term}
+$ mc rb chtc/my-bucket
+```
+
+# 3. Using Staged Files in a Job
+
+## A. Transferring Large Input Files
+
+To use data placed in a CHTC S3 bucket, add files to your submit
+file's `transfer_input_files` that point to the filename
+(e.g. `large-input.file`) inside your bucket (e.g. `my-bucket`) on
+CHTC's S3 storage (`s3dev.chtc.wisc.edu`):
+
+``` {: .file}
```

Comment:
Something wonky is happening w/ the formatting here when I render it locally.

Reply:
<img width="1228" alt="Screen Shot 2020-03-25 at 12 39 02 PM" src="https://user-images.githubusercontent.com/4634021/77567811-b2f7fb00-6e95-11ea-9d22-5ccb3002dbe0.png">

Reply:
Looking back at other pages, it looks like file examples have different (and weirder that I don't understand) syntax.

Reply:
yeah, I often use what the carpentries does, which has the `{}` stuff after the closing backticks. 

What's weird to me is this works further up the page!

Reply:
Fixed in https://github.com/CHTC/chtc-website-source/pull/56/commits/da48907973aec8e31c2468bae7f6080ba410bfc3